platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
 }
 
 - (void)startAnimation {
-	if (active)
-		return;
-	active = TRUE;
-	printf("start animation!\n");
-	if (useCADisplayLink) {
+	// Already active?
+	if (self.isActive) return;
+
+	self.active = YES;
 
-		// Approximate frame rate
-		// assumes device refreshes at 60 fps
-		int frameInterval = (int)floor(animationInterval * 60.0f);
+	printf("start animation!\n");
 
-		displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(drawView)];
-		[displayLink setFrameInterval:frameInterval];
+	if (self.useCADisplayLink) {
+
+		// Create a display link and try and grab the fatest refresh rate
+		_displayLink = [CADisplayLink displayLinkWithTarget:self
+												   selector:@selector(drawView)];
+		if ([_displayLink respondsToSelector:@selector(preferredFramesPerSecond)]) {
+			_displayLink.preferredFramesPerSecond = 0;
+			// We could potentially check the display link's maximumFPS here.
+			// If it supports 120 we could force the preferredFPS to that.
+			// Currently that the's only way to drive a game at 120Hz
+		} else {
+			// Approximate frame rate: assumes device refreshes at 60 fps.
+			// Note that newer iOS devices are 120Hz screens
+			_displayLink.frameInterval = 1;
+		}
 
 		// Setup DisplayLink in main thread
-		[displayLink addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSRunLoopCommonModes];
+		[_displayLink addToRunLoop:[NSRunLoop currentRunLoop]
+						   forMode:NSRunLoopCommonModes];
 	} else {
-		animationTimer = [NSTimer scheduledTimerWithTimeInterval:animationInterval target:self selector:@selector(drawView) userInfo:nil repeats:YES];
+		// This is an extremely terrible way to animate. Very low resolution with lots
+		// of drift. You would be lucky to have this called every 16ms
+		animationTimer = [NSTimer scheduledTimerWithTimeInterval:self.animationInterval
+														  target:self
+														selector:@selector(drawView)
+														userInfo:nil
+														 repeats:YES];
 	}
 
-	if (video_playing) {
-		_unpause_video();
-	}
+	// if (video_playing) {
+	// 	_unpause_video();
```

Comment:
The previous pause logic seemed incorrect and I'd love some feedback. Previously this check would never get hit because it's literally checking if the video is already playing, and if so it calls `play`. Not sure what the intended behavior is.